examples/plot_metric_examples.py

```diff
@@ -0,0 +1,410 @@
+"""
+Algorithms walkthrough
+~~~~~~~~~~~~~~~~~~~~~~
+
+This is a small walkthrough which illustrates all the Metric Learning
+algorithms implemented in metric-learn, and also does a quick
+visualisation which can help understand which algorithm might be best
+suited for you.
+
+Of course, depending on the data set and the constraints your results
+will look very different; you can just follow this and change your data
+and constraints accordingly.
+
+"""
+
+# License: BSD 3 clause
+# Authors: Bhargav Srinivasa Desikan <bhargavvader@gmail.com>
+
+######################################################################
+# Imports
+# ^^^^^^^
+#
+
+from sklearn.manifold import TSNE
+from sklearn.utils import shuffle
+
+import metric_learn
+import numpy as np
+from sklearn.datasets import make_classification
+
+# visualisation imports
+import matplotlib.pyplot as plt
+np.random.seed(42)
+
+
+######################################################################
+# Loading our data-set and setting up plotting
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+# 
+# We will be using the IRIS data-set to illustrate the plotting. You can
+# read more about the IRIS data-set here:
+# `link <https://en.wikipedia.org/wiki/Iris_flower_data_set>`__.
+
+X, Y = make_classification(n_classes=3, n_clusters_per_class=2,
+                           n_informative=3, class_sep=4., n_features=5,
+                           n_redundant=0, shuffle=True)
+
+##########################################################################
+# Let's make the noise of the non-informative features higher, and shuffle
+# the dataset:
+
+X[:, 3:] *= 20
+X, Y = shuffle(X, Y)
+
+
+###########################################################################
+# Note that the dimensionality of the data is 5, so to plot the
+# transformed data in 2D, we will use the t-sne algorithm. (See
+# `sklearn.manifold.TSNE`).
+
+def plot(X, Y):
+    plt.figure(figsize=(8, 6))
+
+    # clean the figure
+    plt.clf()
+
+    tsne = TSNE()
+    X_embedded = tsne.fit_transform(X)
+    plt.scatter(X_embedded[:, 0], X_embedded[:, 1], c=Y, cmap=plt.cm.Paired)
+
+    plt.xticks(())
+    plt.yticks(())
+
+    plt.show()
+
+###################################
+# Let's now plot the dataset as is.
+
+plot(X, Y)
+
+
+######################################################################
+# Metric Learning
+# ^^^^^^^^^^^^^^^
+# 
+# Why is Metric Learning useful? We can, with prior knowledge of which
+# points are supposed to be closer, figure out a better way to understand
+# distances between points. Especially in higher dimensions when Euclidean
+# distances are a poor way to measure distance, this becomes very useful.
+# 
+# Basically, we learn this distance:
+# :math:`D(x,y)=\sqrt{(x-y)\,M^{-1}(x-y)}`. And we learn this distance by
+# learning a Matrix :math:`M`, based on certain constraints.
+# 
+# Some good reading material for the same can be found
```

Comment:
also point to the section "What is Metric Learning"? of the doc

Reply:
I agree, done